From: Takashi Iwai <tiwai@suse.de>
Subject: au88x0 - do codec init more like windows does
Patch-mainline: not yet
References: 

The following patch makes the codec init code act more like the windows
code.

Signed-off-by: Jeff Muizelaar <jeff@infidigm.net>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

--- linus-2.6/sound/pci/au88x0/au88x0_core.c	2005-02-22 10:44:14.000000000 +0100
+++ linux/sound/pci/au88x0/au88x0_core.c	2005-02-23 11:58:49.000000000 +0100
@@ -2480,7 +2480,8 @@ static void vortex_codec_init(vortex_t *
 	int i;
 
 	for (i = 0; i < 32; i++) {
-		hwwrite(vortex->mmio, (VORTEX_CODEC_CHN + (i << 2)), 0);
+		/* the windows driver writes -i, so we write -i */
+		hwwrite(vortex->mmio, (VORTEX_CODEC_CHN + (i << 2)), -i);
 		msleep(2);
 	}
 	if (0) {
@@ -2502,7 +2503,7 @@ static void vortex_codec_init(vortex_t *
 		hwwrite(vortex->mmio, VORTEX_CODEC_CTRL, 0x00e8);
 	}
 	for (i = 0; i < 32; i++) {
-		hwwrite(vortex->mmio, (VORTEX_CODEC_CHN + (i << 2)), 0);
+		hwwrite(vortex->mmio, (VORTEX_CODEC_CHN + (i << 2)), -i);
 		msleep(5);
 	}
 	hwwrite(vortex->mmio, VORTEX_CODEC_CTRL, 0xe8);
